docs/DeveloperGuide.md

```diff
@@ -639,21 +637,97 @@ testers are expected to do more *exploratory* testing.
+      
+**Tips:** Check out the User Guide (UG) for more information on what it means to have duplicate suppliers.
+
+### Finding a supplier
+
+1. Finding a supplier by name
+
+   1. Prerequisites: At least one supplier with the name "John" in the list.
+
+   2. Test case: `find -s n/John`<br>
+      Expected: The suppliers with name containing "John" is shown in the list. Other suppliers are not shown.
+
+   3. Test case: `find -s n/John com/comp`<br>
+      Expected: The suppliers with name containing "John" and company name containing "comp" is shown in the list. Other suppliers are not shown.
+
+   4. Test case: `find -s n/John com/comp pro/rice`<br>
+      Expected: The suppliers with name containing "John", company name containing "comp" and product containing "rice" is shown in the list. Other suppliers are not shown.
+   
+   6. Test case: `find -s n/John com/comp pro/rice pro/bread`<br>
+      Expected: The suppliers with name containing "John", company name containing "comp" and product containing "rice" and "bread" is shown in the list. Other suppliers are not shown.
+
 ### Deleting a supplier
 
 1. Deleting a supplier while all suppliers are being shown
 
-   1. Prerequisites: List all suppliers using the `list` command. Multiple suppliers in the list.
+   1. Prerequisites: List all suppliers using the `list -s` command. Multiple suppliers in the list.
 
    1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+      Expected: First supplier is deleted from the list. Details of the deleted contact shown in the status message.
 
    1. Test case: `delete 0`<br>
-      Expected: No supplier is deleted. Error details shown in the status message. Status bar remains the same.
+      Expected: No supplier is deleted. Error details shown in the status message.
 
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
+### Sorting suppliers
+
+1. Sorting suppliers by name
+
+   1. Prerequisites: List all suppliers using the `list -s` command. Multiple suppliers in the list.
+
+   1. Test case: `sort -s so/a sb/n`<br>
+      Expected: Suppliers are sorted by name in ascending order.
+
+   1. Test case: `sort -s so/d sb/n`<br>
+      Expected: Suppliers are sorted by name in descending order.
+
+   1. Test case: `sort -s sb/n`<br>
+      Expected: Error message is shown. Suppliers are not sorted.
+
+### Listing suppliers
+
+1. Listing all suppliers
+
+   1. Prerequisites: At least one supplier in the list.
+
+   1. Test case: `list -s`<br>
+      Expected: All suppliers are shown in the list.
+
+### Mark supplier status
+
+1. Marking a supplier as active/inactive
+
+   1. Prerequisites: List all suppliers using the `list -s` command. Multiple suppliers in the list.
+
+   1. Test case: `mark -s 1 active`<br>
```

Comment:
Should there be a negative test case for when marking an active supplier as active again?